pkgs/_test_macros/lib/json_codable.dart

```diff
@@ -49,7 +49,28 @@ class JsonCodableImplementation implements Macro {
     final target = request.target;
     final model = await host.query(Query(target: target));
     final clazz = model.uris[target.uri]!.scopes[target.name]!;
-    // TODO(davidmorgan): check for super `fromJson`.
+
+    MacroScope.current.addModel(model);
+    final superclassName = MacroScope.current.typeSystem.supertypeOf(target);
+
+    var superclassHasFromJson = false;
+    if (superclassName.asString != 'dart:core#Object') {
```

Comment:
Just a comment on this API as well, the string comparison here seems less than ideal. It also is going to have issues if for instance an identifier is being re-exported from some other library. We probably need a better way to do this. This type of check is only going to work for core libraries really.